processor/transformprocessor/README.md

```diff
@@ -8,66 +8,146 @@
+
+For each signal type, the processor takes a list of statements associated to a [Context type](#contexts) and executes the statements against the incoming telemetry in the order specified in the config.
+Each statement can access and transform telemetry using functions and allow the use of a condition to help decide whether the function should be executed.
 
 ## Config
 
-The transform processor allows configuring statements for traces, metrics, and logs. Each signal specifies a list of string statements that get passed to the OTTL for interpretation.
+The transform processor allows configuring multiple context statements for traces, metrics, and logs.
+The value of `context` specifies which [OTTL Context](#contexts) to use when interpreting the associated statements.
+The statement strings, which must be OTTL compatible, will be passed to the OTTL and interpreted using the associated context. 
+Each context will be processed in the order specified and each statement for a context will be executed in the order specified.
 
 ```yaml
 transform:
-  <traces|metrics|logs>:
-    statements:
-      - string
-      - string
-      - string
+  <trace|metric|log>_statements:
+    - context: string
+      statements:
+        - string
+        - string
+        - string
+    - context: string
+      statements:
+        - string
+        - string
+        - string
 ```
 
+Proper use of contexts will provide increased performance and capabilities.  See [Contexts](#contexts) for more details.
+
+Valid values for `context` are:
+
+| Signal            | Context Values                                 |
+|-------------------|------------------------------------------------|
+| trace_statements  | `resrouce`, `scope`, `trace`, and `spanevent`  |
+| metric_statements | `resrouce`, `scope`, `metric`, and `datapoint` |
+| log_statements    | `resrouce`, `scope`, and `log`                 |
+
 ## Example
 
+The example takes advantage of context efficiency by grouping transformations with the context which it indents to transform.
+See [Contexts](#contexts) for more details.
+
 Example configuration:
 ```yaml
 transform:
-  traces:
-    statements:
-      - set(status.code, 1) where attributes["http.path"] == "/health"
-      - keep_keys(resource.attributes, ["service.name", "service.namespace", "cloud.region", "process.command_line"])
-      - set(name, attributes["http.route"])
-      - replace_match(attributes["http.target"], "/user/*/list/*", "/user/{userId}/list/{listId}")
-      - replace_pattern(resource.attributes["process.command_line"], "password\\=[^\\s]*(\\s?)", "password=***")
-      - limit(attributes, 100, [])
-      - limit(resource.attributes, 100, [])
+  trace_statements:
+    - context: resource
```

Comment:
I'm a little out of date on this processor, so this may be apparent, but am I right to think that we're using a slice here because it may be necessary to use the same context twice?

```yaml
- context: resource
  ...
- context: trace
  ...
- context: resource 
```

Asking because I don't see any examples or tests that use the same context twice. If it's useful, it might be good to have an example and/or test.

Reply:
Yes that is a feature that is permitted on purpose.  I will add a test for this scenario.

Reply:
@djaglowski new test cases add, thanks for catching that.